roles/docker/templates/rh_docker.repo.j2

```diff
@@ -4,3 +4,4 @@ baseurl=https://yum.dockerproject.org/repo/main/centos/7
 enabled=1
 gpgcheck=1
 gpgkey=https://yum.dockerproject.org/gpg
+{% if http_proxy %}proxy={{ http_proxy }}{% endif %}
```

Comment:
This test will fail actually if http_proxy is undefined. You should write it as 
``` {% if http_proxy is defined %}proxy={{ http_proxy }}{% endif %} ```
Alternately, you can just default it:
``` proxy={{ http_proxy|default('') }}```